File: crates/dewasm-backend-ruby/units/wasi/args_sizes_get.rb
# requires: memory/iws
def wasi_args_sizes_get(argc_ptr, buf_size_ptr)
  @memory.iws(argc_ptr, @args.size)
  @memory.iws(buf_size_ptr, @args.sum { |a| a.bytesize + 1 })
  ERRNO_SUCCESS
end

File: crates/dewasm-backend-perl/units/wasi/args_sizes_get.pl
# requires: memory/i32_store
sub wasi_args_sizes_get {
    my ($self, $argc_ptr, $buf_size_ptr) = @_;
    my $bytes = 0;
    $bytes += length($_) + 1 for @{$self->{args}};
    $self->{memory}->i32_store($argc_ptr, scalar @{$self->{args}});
    $self->{memory}->i32_store($buf_size_ptr, $bytes);
    return ERRNO_SUCCESS;
}
